ui/selectors/selectors.js

```diff
@@ -2982,7 +2984,23 @@ export function getMetaMetricsDataDeletionStatus(state) {
   return state.metamask.metaMetricsDataDeletionStatus;
 }
 
+/**
+ * Gets the remote feature flags from either the manifest or state.
+ * First checks if remote feature flags exist in the manifest and returns those if present.
+ * Otherwise returns the remote feature flags from the MetaMask state that's retrieved from controller.
+ *
+ * @param {object} state - The MetaMask state object
+ * @returns {object} The remote feature flags object containing feature flag key-value pairs
```

Comment:
The call to `getManifestFlags` returns a proper type, so we should return it instead of `object`.

Also, since you are already adding types to this function it'd be easy to move this function (and its tests) to typescript files.

Reply:
We do have a task to refactor selector.js to typescript https://github.com/MetaMask/metamask-extension/issues/29628, I think it's better to leave until then (which is soon)
On the other side, I agree we should include better return type in this function, modified this part in 473a3bd2b5ff63be8da0f6a452d76edd371af1d0

Reply:
The changes you linked to is not in the current HEAD of the branch

Reply:
it shows correctly for me, can you double check?
<img width="1592" alt="Screenshot 2025-01-17 at 17 35 31" src="https://github.com/user-attachments/assets/80125f30-eabf-4d45-afb5-4738ba253d72" />

Reply:
In fact, Howard proposed the same and now it is making more sense to me as well to move it out, sorry i didn't get it clear earlier. Add in 5308bed8ec062195685cd4199d01f306ff422009 and 2cc15607d22f8ff797ff7827b7bc9e02564aafa4